src/esrs/state.rs

```diff
@@ -25,21 +51,21 @@ impl<S: Default> AggregateState<S> {
     /// might happen.
     ///
     /// Prefer [Default] implementation.
-    #[must_use]
     pub fn new(id: impl Into<Uuid>) -> Self {
         Self {
             id: id.into(),
             inner: Default::default(),
             sequence_number: 0,
+            lock: None,
```

Comment:
The reason for this is that when you create a brand new AggregateState, it *should* have a different `id` than any other aggregate you already have, so you should not incur into optimistic locking issues.
This is indeed true for `Default`. However nothing stops you from `AggregateState::new(used_id)`, which would probably cause optimistic locking issues anyway I guess.